platform/security/security-migratable/src/main/java/org/codice/ddf/security/migratable/impl/SecurityMigratable.java

```diff
@@ -39,74 +31,83 @@
-            exportCrlFile(propertiesPath, exportDirectory, migrationWarnings);
-        }
+    @Override
+    public String getTitle() {
+        return "Security Migration";
     }
 
-    private void exportCrlFile(Path propertiesPath, Path exportDirectory,
-            Collection<MigrationWarning> migrationWarnings) throws MigrationException {
-        LOGGER.debug("Exporting CRL from property [{}] in file [{}]...",
-                CRL_PROP_KEY,
-                propertiesPath.toString());
-        String crlPathStr = migratableUtil.getJavaPropertyValue(propertiesPath, CRL_PROP_KEY);
-
-        if (crlPathStr == null) {
-            return;
-        }
+    @Override
+    public String getDescription() {
+        return "Exports Security system files";
+    }
 
-        if (StringUtils.isWhitespace(crlPathStr)) {
-            String error = String.format(
-                    "Failed to export CRL. No CRL path found in file [%s]. Property [%s] from properties file [%s] has a blank value.",
-                    propertiesPath,
-                    CRL_PROP_KEY,
-                    propertiesPath);
-            throw new ExportMigrationException(error);
-        }
+    @Override
+    public String getOrganization() {
+        return "Codice";
+    }
 
-        Path crlPath = Paths.get(crlPathStr);
-        migratableUtil.copyFile(crlPath, exportDirectory, migrationWarnings);
+    @Override
+    public void doExport(ExportMigrationContext context) {
+        SecurityMigratable.PROPERTIES_FILES.stream()
+                .map(context::getEntry)
+                .peek(me -> LOGGER.debug("Exporting CRL from property [{}] in file [{}]...",
+                        SecurityMigratable.CRL_PROP_KEY,
+                        me.getPath()))
+                // do not automatically record an error if property is not defined (just skip that file)
+                .map(me -> me.getPropertyReferencedEntry(SecurityMigratable.CRL_PROP_KEY,
+                        (r, v) -> (v != null)))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .forEach(ExportMigrationEntry::store);
+        LOGGER.debug("Exporting PDP files from [{}]...", SecurityMigratable.PDP_POLICIES_DIR);
+        context.entries(SecurityMigratable.PDP_POLICIES_DIR)
+                .forEach(ExportMigrationEntry::store);
     }
 
-    private void exportPdpDirectory(Path exportDirectory,
-            Collection<MigrationWarning> migrationWarnings) throws MigrationException {
-        LOGGER.debug("Exporting PDP Directory at [{}]...", PDP_POLICIES_DIR.toString());
-        migratableUtil.copyDirectory(PDP_POLICIES_DIR, exportDirectory, migrationWarnings);
+    @Override
+    public void doImport(ImportMigrationContext context) {
+        SecurityMigratable.PROPERTIES_FILES.stream()
+                .map(context::getEntry)
+                .peek(me -> LOGGER.debug("Importing CRL from property [{}] in file [{}]...",
```

Comment:
✏️ Same comment about over-logging and potentially switching to a loop.

Reply:
Same reply :-)